Moblin/HaishinKit/Media/VideoUnit.swift

```diff
@@ -41,20 +41,31 @@ private struct FaceDetectionsCompletion {
 
 private class ReplaceVideo {
     private var latency: Double
+    private var frameRate: Double
     private var sampleBuffers: Deque<CMSampleBuffer> = []
+    private var realPresentationTimeStamp: CMTime = .zero
     private var firstPresentationTimeStamp: Double = .nan
     private var currentSampleBuffer: CMSampleBuffer?
 
-    init(latency: Double) {
+    init(latency: Double, frameRate: Double) {
         self.latency = latency
+        self.frameRate = frameRate
+        self.realPresentationTimeStamp = CMClockGetTime(CMClockGetHostTimeClock())
     }
 
     func appendSampleBuffer(_ sampleBuffer: CMSampleBuffer) {
         sampleBuffers.append(sampleBuffer)
         sampleBuffers.sort { $0.presentationTimeStamp < $1.presentationTimeStamp }
     }
 
-    func updateSampleBuffer(_ realPresentationTimeStamp: Double) {
+    func updateSampleBuffer() {
+        realPresentationTimeStamp = CMTimeAdd(
+            realPresentationTimeStamp,
+                   CMTime(
+                       value: CMTimeValue(Double(1)),
+                       timescale: CMTimeScale(frameRate)
+                   ))
```

Comment:
I think it is better tot calculate the PTS from a frame counter for less drift.

Reply:
builtin camera should now work after background/foreground with latest commit